src/vault.py

```diff
@@ -385,20 +391,23 @@ def value_from_path(secret_data, path):
 
 def dict_walker(pattern, data, args, envs, secret_data, path=None):
     # Walk through the loaded dicts looking for the values we want
-    path = path if path is not None else ""
+    environment = f"/{args.environment}" if args.environment is not None else ""
+    path = path if path is not None else environment
     action = args.action
     if isinstance(data, dict):
         for key, value in data.items():
             if value == pattern or str(value).startswith(envs[VAULT_TEMPLATE_POSITION]):
                 if value.startswith(envs[VAULT_TEMPLATE_POSITION]):
-                    _full_path = value[len(envs[VAULT_TEMPLATE_POSITION]):]
+                    _full_path = value[len(envs[VAULT_TEMPLATE_POSITION]):].replace("{environment}", environment)
                 else:
                     _full_path = None
                 if action == "enc":
+                    path_sans_env = path.replace(environment, '')
                     if secret_data:
-                        data[key] = value_from_path(secret_data, f"{path}/{key}")
+                        data[key] = value_from_path(secret_data, f"{path_sans_env}/{key}")
                     else:
-                        data[key] = input(f"Input a value for {path}/{key}: ")
+                        path_to_property_syntax = path_sans_env.replace("/", ".")[1:]
+                        data[key] = input(f"Input a value for {path_to_property_syntax}.{key}: ")
```

Comment:
hey just wanna point out this change b/c I made it kinda on the fly

Changes the output to be `Input a value for my.property` instead of `/my/property` because I felt like it was more clear that it relates to the yaml property in question, not the path in helm.

Reply:
Yea, that makes sense.